mixed-content/generic/common.js

```diff
@@ -313,10 +318,18 @@ function createMediaElement(type, media_attrs, source_attrs) {
   var sourceElement = createElement("source", {});
 
   mediaElement.eventPromise = new Promise(function(resolve, reject) {
-    mediaElement.addEventListener("loadeddata", resolve);
+    mediaElement.addEventListener("loadeddata", function (e) {
+      resolve(e);
+    });
+
+    // Safari doesn't fire an `error` event when blocking mixed content.
```

Comment:
What behavior is required by specs, and is there another test that'd still be failing in Safari assuming Safari's isn't the right behavior?

Reply:
> What behavior is required by specs,

The spec turns mixed content blocking into a synthetic network error. So, the element should react in whatever way it reacts to any other network error. I think `error` is correct, and I filed https://bugs.webkit.org/show_bug.cgi?id=179116 to follow up on it.

> and is there another test that'd still be failing in Safari assuming Safari's isn't the right behavior?

I don't understand the question, can you clarify?

Reply:
What I mean is, is there now a test that's failing because of https://bugs.webkit.org/show_bug.cgi?id=179116 and that will start passing in WebKit if the bug is fixed?

We're building stuff for wpt.fyi to highlight tests that are failing in "your" browser but passing in some/most/all others, so making sure there are failing tests will hopefully, eventually, be a way of getting interop bugs like this higher up on the radar of the team who needs to make a change.